core/background_workers.py

```
@@ -0,0 +1,433 @@
+    
+    while True:
+        try:
+            # Get interval from settings (dynamic)
+            interval_str = await get_setting_value(pool, 'local_sync_interval_seconds', '120')
+            interval = int(interval_str)
+            
+            # Scan for queue_input_sms keys
+            keys = await redis_pool.scan("queue_input_sms:*")
+            
+            if keys:
+                logger.info(f"Dumping {len(keys)} SMS records to PostgreSQL")
```

Ruff (0.14.5)

Logging statement uses f-string

(G004)